crates/ruff_python_codegen/src/generator.rs

```diff
@@ -1912,6 +1936,12 @@ class Foo:
         assert_round_trip!(r#"f"{ chr(65)  =   :#x}""#);
         assert_round_trip!(r#"f"{  ( chr(65)  ) = }""#);
         assert_round_trip!(r#"f"{a=!r:0.05f}""#);
+        assert_round_trip!(&r#"
+f"{1=
+}"
+"#
+            .trim()
+            .replace('\n', LineEnding::default().as_str())); // https://github.com/astral-sh/ruff/issues/18742
```

Comment:
```suggestion
            ); // https://github.com/astral-sh/ruff/issues/18742
```

`assert_roundtrip` already replaces the newlines with the platform's defualt.

Reply:
The failing test after removing all the `LineEnding` conversions before calling `assert_round_trip` looks correct to me. It reveals that the round-tripped code uses `\n` instead of windows line endings. 

I think the reason for this is that `debug_text` gets normalized by the parser. Meaning, ut sues `\n` everywhere.

Reply:
I figured out part of what was going wrong, I had a bunch of duplicate tests everywhere due to coding on my phone. 
I don't think that's the case? Since in the original issue on the playground which uses CRLF endings the code was ending up as `\\r\\n`